src/javascript/ContentEditor/EditPanel/HeaderBadges/ReadOnlyBadge.jsx

```diff
@@ -10,7 +10,7 @@ export const ReadOnlyBadge = ({isGlobal, readOnly}) => {
     const {nodeData} = useContentEditorContext();
     const {sections} = useContentEditorSectionContext();
 
-    const hasNotReadOnlySection = sections.find(s => s.fieldSets.find(fs => !fs.readOnly));
+    const hasNotReadOnlySection = sections.find(s => s.fieldSets.find(fs => fs.fields.find(f => !f.readOnly)));
```

Comment:
should we also still keep the check on `fs.readOnly`? I'm not sure if there's a scenario where all fields are not read only but the fieldset itself is set to readOnly?